### PR TITLE
Change "Clear" button behavior on mobile sidebar

### PR DESCRIPTION
**Pull request by iaronaraujo**
#### What problem is this solving?

Previously, on a mobile search view, when clicking on the "clear" button of the filter sidebar, only the checkboxes that were checked before clicking on "Apply" would be cleaned. This behavior confused many of our clients, so we decided to change this button to clear everything that is not:
- Category
- Department
- Collection (clusterId)
- Price Range

#### How to test it?

[storecomponents](https://iaronaraujo--storecomponents.myvtex.com/apparel---accessories/samsung?map=b)
[als](https://iaronaraujo--alssports.myvtex.com/clothing/Mens?map=c,c,c&query=/Clothing/Accessories/Hats)
[motorolaus](https://iaronaraujo--motorolaus.myvtex.com/x?_q=x&map=ft)
[exitocol](https://iaronaraujo--exitocol.myvtex.com/tecnologia/televisores)
[eriksbikes](https://iaronaraujo--eriksbikeshop.myvtex.com/bicycles/cycling/sale?fuzzy=0&map=category,department,deal&operator=and)
[wwg](https://iaronaraujo--worldwidegolf.myvtex.com/ball/mens?_q=ball&fuzzy=0&map=ft,gender&operator=and)
[lionspride](https://iaronaraujo--lionspride.myvtex.com/l/shirts?map=size,category)

#### Screenshots or example usage:

Jun-18-2020 15-28-16

### CODE REVIEW

File: react/components/FilterSidebar.js [hunk 1]
```diff
@@ -74,21 +83,40 @@ const FilterSidebar = ({
...
   }
 
+  const isCategoryDepartmentCluster = term => {
+    return (
+      term === MAP_CATEGORY_CHAR ||
+      term === PRODUCT_CLUSTER_IDS ||
+      term === CATEGORY ||
+      term === DEPARTMENT
+    )
+  }
```
Comment (klzns) [inline react/components/FilterSidebar.js:93]: That is a pure function. We can move this to be defined outside of the component so we don't create it a new function on every render.

Also, the name and the parameter name are not clear. 

You could do something like

```js
const SEARCH_PARAMS = [MAP_CATEGORY_CHAR, PRODUCT_CLUSTER_IDS, CATEGORY, DEPARTMENT]

const tanana = term => SEARCH_PARAMS.includes(term)
```

File: react/components/FilterSidebar.js [hunk 2]
```diff
@@ -97,18 +125,43 @@ const FilterSidebar = ({
...
+        if (isCategoryDepartmentCluster(map[i])) {
+          newQuery.push(query[i])
+          newMap.push(map[i])
+        }
+      }
+      return {
+        query: newQuery.join(PATH_SEPARATOR),
+        map: newMap.join(MAP_VALUES_SEP),
+      }
+    }
```
Comment (klzns) [inline react/components/FilterSidebar.js:149]: Move this pure function outside of the component

File: react/hooks/useSelectedFilters.js [hunk 3]
```diff
@@ -20,12 +21,26 @@ const useSelectedFilters = facets => {
...
+  const checkMap = (slugMap, facetMap) => {
+    if (slugMap === MAP_CATEGORY_CHAR) {
+      return (
+        facetMap === MAP_CATEGORY_CHAR ||
+        facetMap === DEPARTMENT ||
+        facetMap === CATEGORY
+      )
+    }
+    return slugMap === facetMap
+  }
```
Comment (klzns) [inline react/hooks/useSelectedFilters.js:35]: Move outside

File: react/components/FilterSidebar.js [hunk 4]
```diff
@@ -97,18 +125,43 @@ const FilterSidebar = ({
...
+    are important to show the correct facets. */
+    if (shouldClear.current) {
+      shouldClear.current = false
+      return filterCategoryDepartmentCluster(filterContext)
+    }
+
     return {
       ...filterContext,
-      ...buildNewQueryMap(fullText, filterOperations, selectedFilters),
+      ...buildNewQueryMap(fullText, filterOperations, [...selectedFilters]),
```
Comment (Klynger) [inline react/components/FilterSidebar.js:164]: Why this?

Comment (iaronaraujo) [reply react/components/FilterSidebar.js:137]: Added comment explaining this!

**Pull request merged**